actions/comment/save.php

```diff
@@ -11,44 +11,42 @@
 $comment_text = get_input('generic_comment');
 
 if (empty($comment_text)) {
-	register_error(elgg_echo("generic_comment:blank"));
-	forward(REFERER);
+	return elgg_error_response(elgg_echo('generic_comment:blank'));
 }
 
 if ($comment_guid) {
 	// Edit an existing comment
 	$comment = get_entity($comment_guid);
 
 	if (!elgg_instanceof($comment, 'object', 'comment')) {
-		register_error(elgg_echo("generic_comment:notfound"));
-		forward(REFERER);
+		return elgg_error_response(elgg_echo('generic_comment:notfound'));
 	}
 	if (!$comment->canEdit()) {
-		register_error(elgg_echo("actionunauthorized"));
-		forward(REFERER);
+		return elgg_error_response(elgg_echo('actionunauthorized'));
 	}
 
 	$comment->description = $comment_text;
-	if ($comment->save()) {
-		system_message(elgg_echo('generic_comment:updated'));
+	if (!$comment->save()) {
+		return elgg_error_response(elgg_echo('generic_comment:failure'));
+	}
+	
 
-		if (elgg_is_xhr()) {
-			// @todo move to its own view object/comment/content in 1.x
-			echo elgg_view('output/longtext', [
-				'value' => $comment->description,
-				'class' => 'elgg-inner',
-				'data-role' => 'comment-text',
-			]);
-		}
-	} else {
-		register_error(elgg_echo('generic_comment:failure'));
+	if (elgg_is_xhr()) {
+		// @todo move to its own view object/comment/content in 1.x
+		echo elgg_view('output/longtext', [
+			'value' => $comment->description,
+			'class' => 'elgg-inner',
+			'data-role' => 'comment-text',
+		]);
 	}
+	
+	system_message(elgg_echo('generic_comment:updated'));
```

Comment:
Yes, you can just put plain HTML as the first argument in `elgg_ok_response()`. It's smart enough to handle both scalar and non-scalar values.